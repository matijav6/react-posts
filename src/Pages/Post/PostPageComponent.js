import React from 'react';
import {BackButton} from '../../components/BackButton/BackButton';
import {CustomSpinner} from '../../components/Spinner/Spinner';
import {Post} from '../../components/Post/Post';
import {Comment} from '../../components/Comment/Comment';
import PropTypes from 'prop-types';

export const PostPageComponent = ({postData, isLoading}) => {
        return (
            <>  <CustomSpinner isLoading={isLoading}/>
                <nav className="navbar navbar-dark bg-primary d-flex justify-content-center min-height-50"/>
                <div className={'container-fluid p-3'}>
                    <div className={'row justify-content-md-center'}>
                        <div className={'col-sm-8 mb-4 d-flex justify-content-center'}>
                            <div className={'w-100'}>
                                <Post classes={'h-auto mw-100'}>
                                    <Post.Body>
                                        <Post.Title>{postData.post.title}</Post.Title>
                                        <Post.Text>{postData.post.body}</Post.Text>
                                    </Post.Body>
                                </Post>
                                {
                                    postData.comments.map(comment => (
                                        <Comment classes={'w-50 mw-100'} key={comment.id}>
                                            <Comment.Body>
                                                <Comment.Title>{comment.name}</Comment.Title>
                                                <Comment.Text>{comment.body}</Comment.Text>
                                                <Comment.Author>{comment.email}</Comment.Author>
                                            </Comment.Body>
                                        </Comment>
                                    ))
                                }
                            </div>
                        </div>
                    </div>
                </div>
                <BackButton/>
            </>
        );
    }
;
PostPageComponent.propTypes = {
    postData: PropTypes.shape({
        post:  PropTypes.shape({
            body: PropTypes.string.isRequired,
            title: PropTypes.string.isRequired,
        }),
        comments: PropTypes.arrayOf(
            PropTypes.shape({
                body: PropTypes.string.isRequired,
                name: PropTypes.string.isRequired,
                id: PropTypes.number.isRequired,
            }),
        ).isRequired,
    }),
    isLoading: PropTypes.bool.isRequired,
};
