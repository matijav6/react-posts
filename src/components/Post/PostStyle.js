import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  min-width: 0;
  max-width: 450px;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 0.25rem;
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
  overflow: hidden;
  height: 100%;
`;

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
`;

export const Title = styled.h1`
  font-size: 1.25rem;
  font-weight: bold;
  margin: 0;
  max-width: 65%;
`;

export const Text = styled.p`
  line-height: 25px;
  color: #6c757d;
`;

export const TotalComments = styled.p`
  color: #565656;
  position: absolute;
  bottom: 0;
  left: 0;
  border-top: 1px solid gray;
  width: 100%;
  margin: 0;
  padding: 0 10px;
`;

export const User = styled.p`
  color: #565656;
`;
