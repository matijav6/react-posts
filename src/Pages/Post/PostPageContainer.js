import React, {useEffect} from 'react';
import {PostPageComponent} from './PostPageComponent';
import withRequest from '../../helpers/withRequest';
import PropTypes from 'prop-types';

const COMPONENT_NAME = 'Post';

const PostPageContainer = ({hello, location}) => {

    useEffect(() => {
        console.log(`${hello} ${COMPONENT_NAME}`);
        //eslint-disable-next-line
    }, []);

    return (
        <PostPageComponent
            postData={location.state.post}
        />
    );
};

export default withRequest(PostPageContainer);

PostPageContainer.propTypes = {
    hello: PropTypes.string.isRequired,
    location: PropTypes.exact({
        pathname: PropTypes.string,
        search: PropTypes.string,
        hash: PropTypes.string,
        key: PropTypes.string,
        state: PropTypes.exact({
            post: PropTypes.shape({
                body: PropTypes.string.isRequired,
                title: PropTypes.string.isRequired,
                id: PropTypes.number.isRequired,
                userId: PropTypes.number.isRequired,
            }).isRequired,
        }),
    }).isRequired
};
