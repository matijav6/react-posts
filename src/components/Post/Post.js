import React from 'react';
import classNames from 'classnames';
import { Container, Body, Title, Text, TotalComments, User } from './PostStyle';

export const Post = ({classes, children, ...restProps}) => {
    return (
        <Container className={classNames('card', classes)} {...restProps}>
            {children}
        </Container>
    );
};


Post.Body = function CardBody({ classes, children, ...restProps }) {
    return (
        <Body className={classNames('card__body', classes)} {...restProps}>
            {children}
        </Body>
    );
};

Post.Title = function CardTitle({ classes, children, ...restProps }) {
    return (
        <Title className={classNames('card__title', classes)} {...restProps}>
            {children}
        </Title>
    );
};

Post.Text = function CardText({ classes, children, ...restProps }) {
    return (
        <Text className={classNames('card__text', classes)} {...restProps}>
            {children}
        </Text>
    );
};

Post.User = function CardUser({ classes, children, ...restProps }) {
    return (
        <User className={classNames('card__user', classes)} {...restProps}>
            {children}
        </User>
    );
};

Post.TotalComments = function CardTotalCount({ classes, children, ...restProps }) {
    return (
        <TotalComments className={classNames('card__totalComments', classes)}{...restProps}>
            {children}
        </TotalComments>
    );
};
