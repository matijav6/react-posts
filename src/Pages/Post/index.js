import PostPageContainer from './PostPageContainer'

export {PostPageContainer as PostPage}
