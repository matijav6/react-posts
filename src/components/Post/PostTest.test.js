import renderer from 'react-test-renderer';
import {Post} from './Post'

test('Post renders correctly', () => {
    const tree = renderer
        .create(<Post/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
