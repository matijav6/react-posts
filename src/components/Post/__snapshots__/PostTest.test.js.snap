// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders correctly 1`] = `
<div
  className="sc-bdfBwQ fUVpNk card"
/>
`;
